src/VisualStudio/Core/Def/ProjectSystem/OpenTextBufferProvider.cs

```diff
@@ -91,9 +99,27 @@ private void RaiseEventForEachListener(Action<IOpenTextBufferEventListener> acti
         }
     }
 
-    private async Task CheckForExistingOpenDocumentsAsync(IThreadingContext threadingContext)
+    private async Task CheckForExistingOpenDocumentsAsync()
     {
-        await threadingContext.JoinableTaskFactory.SwitchToMainThreadAsync();
+        _threadingContext.ThrowIfNotOnBackgroundThread();
```

Comment:
I don't think this is safe -- if this type _were_ to get created on the UI thread, then this will throw prior to the await Task.Yield. This should just be OK to remove, but you may want to have an alwaysYield on the SwitchToMainThreadAsync().

Reply:
I see. So, your concern is that someone outside of our package load will get to creating this object first? I can go ahead and remove the check as we're currently fine regardless of what thread we're called on.

Reply:
The general rule of MEF is don't ever assert threads in the constructor, since it can be too unpredictable when things create. Obviously we have legacy stuff you're fixing, but we shouldn't add new stuff in. We're probably fine in this case because there's no other explicit MEF import of MiscellaneousFilesWorkspace in the repository, but a random change that were to add one could break this.